alf/optimizers/optimizers.py

```diff
@@ -501,6 +507,28 @@ def load_state_dict(self, state_dict):
                         state['rng_state'] = state['rng_state'].to(
                             self._rng_state_device).byte()
 
+    @common.add_method(NewCls)
+    def __setstate__(self, state):
+        def _move_key_as_the_first(d, key):
+            return {key: d.pop(key)}
+
+        for param_group in self.param_groups:
+            for p in param_group['params']:
+                p_state = self.state[p]
+                if 'step' not in p_state:
+                    # optimizers in higher version torch requires 1) the presence of 'step'
+                    # in state, and 2) it appears as the first key of the state dictionary.
+                    # Therefore we explicitly create it if it does not exist and make it
+                    # the key that comes first
+                    p_state.update({'step': 0})
```

Comment:
Is this needed by all optimizers?

Reply:
Some of the optimizers have this requirement, such Adam, AdamW, ASGD. Here we handle it in a general way without differentiating different variations while without too much additional overhead. Updated comment